File: snake_case/snake_case.py
#!/usr/bin/env python3

def snake_case(chaine):
    if type(chaine) is str:
        resultat = ''
        flag = False

        if chaine.isalpha():
            if chaine[0].isupper():
                return "La chaine n'est pas en camelcase"

            for c in chaine:
                if c.isupper():
                    flag = True

            if flag == True:
                for caractere in chaine:
                    if caractere.isupper():
                        resultat += '_' + caractere.lower()
                    else:
                        resultat += caractere
                return resultat + ' ' + chaine

    return "La chaine n'est pas en camelcase"


# tableau de test
tests = ["salutLaTeam", "salut_la_team", "salut_la_Team", "SalutLaTeam", "salutlateam", "salut la team", "salut_lateam", "salut_la_team_",
         "_salut_la_team", "salut_la_team_", "salut__la_team", "salut_la__team", "salut_la_team_", "salut_la_team_", "salut_la_team_"]


for test in tests:
    print(snake_case(test))

File: snake_case/snake_case.js
function snake_case(chaine) {
  if (typeof chaine === 'string') {
    const miniscules = 'abcdefghijklmnopqrstuvwxyz';
    const majuscules = miniscules.toUpperCase();
    let resultat = '';

    for (const caractere of chaine) {
      if (!(miniscules + majuscules).includes(caractere)) {
        return "la chaine n'est pas en camelcase";
      }

      const indice = chaine.indexOf(caractere);

      if (indice === 0) {
        if (majuscules.includes(caractere)) {
          return "la chaine n'est pas en camelcase";
        }
        resultat += caractere;
      } else {
        if (majuscules.includes(caractere)) {
          resultat += `_${caractere.toLowerCase()}`;
        } else {
          resultat += caractere;
        }
      }
    }

    return resultat;
  }

  return "la chaine n'est pas en camelcase";
}

tests = [
  'salutLaTeam',
  'salut_la_team',
  'salut_la_Team',
  'SalutLaTeam',
  'salutlateam',
  'salut la team',
  'salut_lateam',
  'salut_la_team_',
  '_salut_la_team',
  'salut_la_team_',
  'salut__la_team',
  'salut_la__team',
  'salut_la_team_',
  'salut_la_team_',
  'salut_la_team_',
];

for (const test of tests) {
  console.log(snake_case(test));
}
